src/effects/effectchainslot.cpp

```diff
@@ -116,10 +144,400 @@ EffectChainSlot::~EffectChainSlot() {
+        pChain->m_pEffectsManager, pChain->id(), pChain);
+    pClone->setName(pChain->name());
+    // Do not set the state of the chain because that information belongs
+    // to the EffectChainSlot. Leave that to EffectChainSlot::loadEffectChain.
+    for (const auto& pEffect : pChain->effects()) {
+        EffectPointer pClonedEffect;
+        if (pEffect == nullptr) {
+            // Insert empty effect to preserve chain order
+            pClonedEffect = EffectPointer();
+        } else {
+            pClonedEffect = pChain->m_pEffectsManager->instantiateEffect(
+                    pEffect->getManifest()->id());
+        }
+        pClone->addEffect(pClonedEffect);
+    }
+    return EffectChainPointer(pClone);
+}
+// EffectChainSlotPointer EffectChainSlot::clone(EffectChainSlotPointer pChainSlot) {
+//     if (!pChainSlot) {
+//         return EffectChainSlotPointer();
+//     }
+
+//     EffectChain* pClone = new EffectChainSlot(
+//         pChainSlot->m_pEffectsManager, pChainSlot->id(), pChainSlot);
+//     pClone->setName(pChain->name());
+//     // Do not set the state of the chain because that information belongs
+//     // to the EffectChainSlot. Leave that to EffectChainSlot::loadEffectChain.
+//     for (const auto& pEffect : pChain->effects()) {
+//         EffectPointer pClonedEffect;
+//         if (pEffect == nullptr) {
+//             // Insert empty effect to preserve chain order
+//             pClonedEffect = EffectPointer();
+//         } else {
+//             pClonedEffect = pChain->m_pEffectsManager->instantiateEffect(
+//                     pEffect->getManifest()->id());
+//         }
+//         pClone->addEffect(pClonedEffect);
+//     }
+//     return EffectChainPointer(pClone);
+// }
+
+bool EffectChainSlot::isEmpty() const {
+    return (m_id == QString());
+}
+
+// EffectRack* EffectChainSlot::rack() const {
+//     return m_pEffectRack;
+// }
+
+EffectChainPointer EffectChainSlot::prototype() const {
+    return m_pPrototype;
+}
+
+// const QString& EffectChainSlot::id() const {
+//     return m_id;
+// }
+
+const QString& EffectChainSlot::name() const {
+    return m_name;
+}
+
+void EffectChainSlot::setName(const QString& name) {
+    m_name = name;
+    // emit(nameChanged(name));
+    slotChainNameChanged(name);
+}
+
+QString EffectChainSlot::description() const {
+    return m_description;
+}
+
+void EffectChainSlot::setDescription(const QString& description) {
+    m_description = description;
+    // emit(descriptionChanged(description));
+}
+
+bool EffectChainSlot::enabled() const {
+    return m_bEnabled;
+}
+
+void EffectChainSlot::setEnabled(bool enabled) {
+    m_bEnabled = enabled;
+    sendParameterUpdate();
+    // emit(enabledChanged(enabled));
+    slotChainEnabledChanged(enabled);
+}
+
+void EffectChainSlot::enableForInputChannel(const ChannelHandleAndGroup& handle_group) {
+    // TODO(Be): remove m_enabledChannels from this class and move this logic
+    // to EffectChainSlot
```

Comment:
Let's take care of this TODO. The `updateRoutingSwitches` function should be removed. That function was an ugly hack that was only necessary because of the split between EffectChain and EffectChainSlot.